core/src/main/java/org/elasticsearch/action/get/MultiGetRequest.java

```diff
@@ -327,6 +328,13 @@ public MultiGetRequest add(@Nullable String defaultIndex, @Nullable String defau
                     parseDocuments(parser, this.items, defaultIndex, defaultType, defaultFields, defaultFetchSource, defaultRouting, allowExplicitIndex);
                 } else if ("ids".equals(currentFieldName)) {
                     parseIds(parser, this.items, defaultIndex, defaultType, defaultFields, defaultFetchSource, defaultRouting);
+                } else {
+                    final String message = String.format(
+                            Locale.ROOT,
+                            "Unknown key [%s] for a %s, expected [docs] or [ids]",
+                            currentFieldName,
+                            token);
+                    throw new ParsingException(parser.getTokenLocation(), message);
                 }
             }
```

Comment:
should we also add another else where we throw error if the token is not field_name nor start_object?